src/ctap2/commands/get_assertion.rs

```diff
@@ -121,10 +218,14 @@ impl Serialize for HmacSecretExtension {
         S: Serializer,
     {
         if let Some(calc) = &self.calculated_hmac {
-            let mut map = serializer.serialize_map(Some(3))?;
+            let mut map =
+                serializer.serialize_map(Some(3 + self.pin_protocol.map(|_| 1).unwrap_or(0)))?;
```

Comment:
I started to address this, but then quickly found myself thinking it would be much better to not have this problem in the first place: https://github.com/mozilla/authenticator-rs/pull/338 :smile: